### ISSUE TITLE
DevOps should include PowerShell / DSC

### ISSUE THREAD

**Issue by Xainey** [kamranahmedse/developer-roadmap#36]
- Powershell Desired State Configuration (DSC) should be listed under **DevOps > Automation**.
- Powershell should possibly be included under **DevOps > Love For Terminal** now that it cross platform and trending stronger than ever. Personally, I think this should go under **Required for any path** if developing on a windows machine.

**Comment by beaudryj** [comment]
No NewRelic Under Monitoring and Alerting 
No Windows under Operating Systems 
No Azure Container Services under cluster managers
No MSBuild under Command Tools

**Comment by mgilank** [comment]
+
no backup procedure
no learning for disaster recovery

**Comment by ppihus** [comment]
Isn't bash on Windows way more popular than Powershell on Linux?

Please correct me if I'm wrong but I believe *web* development doesn't usually utilize any of the above mentioned tools and technologies.

**Comment by beaudryj** [comment]
It does if you are running the build chains for said web developers

**Comment by Xainey** [comment]
@ppihus As long as you reference the proper binaries in your system path you can use tools like ruby, gem, npm, etc directly from a PowerShell prompt. Additionally, with PowerShell, you have access to use modules and scripts from public or private repos (such as the PowerShellGallery). You can still use custom scripts, aliases etc in your PS $profile(s) (this could be compared to your bashrc or bash_profile.) For windows developers, PS is a great tool for configuring, querying, and scaffolding stuff in the windows ecosystem - These are especially great for IIS, Azure, and Active Directory as @beaudryj  mentioned. 

In the past, I used git bash, but I find that with PowerShell and the new WSL I have much more flexibility as opposed to emulated tools like cygwin. I'm not claiming to be an expert but I have had experience in front-end/back-end development before shifting focus to DevOps.

**Comment by esuiru** [comment]
Agreed, Powershell should be added in the DevOps road map for the reasons @Xainey and @beaudryj mentioned. Anyone deploying Windows, or using Azure, Powershell is the default and built in tool for administration and automation. 

Powershell is also multi platform, being able to run on OSX and *Nix. While Bash can also run on Windows, if you're dealing in a Windows Environment and interacting w/ Azure. 

Especially if you're in an Windows IIS/AD (ADFS)/Azure environment.
